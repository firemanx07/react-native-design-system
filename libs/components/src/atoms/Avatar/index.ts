export { default as Avatar, AvatarSize } from './Avatar';
