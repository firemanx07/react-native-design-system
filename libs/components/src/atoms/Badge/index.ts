export { default as Badge, BadgeSize, BadgeVariant } from './Badge';
