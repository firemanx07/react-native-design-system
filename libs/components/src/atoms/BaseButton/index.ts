export { default as BaseButton } from './BaseButton';
export { ButtonVariant, ButtonSize, ButtonShape } from './BaseButton.types';
