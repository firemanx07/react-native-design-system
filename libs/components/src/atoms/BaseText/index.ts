export { default as BaseText, TextVariant, TextWeight } from './BaseText';
