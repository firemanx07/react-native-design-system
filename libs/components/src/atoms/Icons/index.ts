export * from './svg';
export * from './pdp';
