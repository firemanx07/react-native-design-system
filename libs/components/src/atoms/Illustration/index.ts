export {
  default as Illustration,
  IllustrationSize,
  IllustrationName,
} from './Illustration';
