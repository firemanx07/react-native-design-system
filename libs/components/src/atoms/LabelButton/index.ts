export { default as LabelButton, LabelButtonVariant } from './LabelButton';
