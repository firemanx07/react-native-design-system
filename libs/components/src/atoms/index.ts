export * from './BaseText';
export * from './BaseButton';
export * from './Checkbox';
export * from './Circle';
export * from './LabelButton';
export * from './IconButton';
export * from './Icons';
export * from './AppIcon';
export * from './Avatar';
export * from './Badge';
export * from './Illustration';
export * from './LoadingIndicator';
export * from './Logo';
export * from './RadioButton';
export * from './Switch';
