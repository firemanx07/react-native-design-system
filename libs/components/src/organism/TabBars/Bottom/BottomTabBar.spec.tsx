import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { NavigationContainer } from '@react-navigation/native';
import React from 'react';

import { renderTabs, testTabs } from '../../../../test-utils/test-tabs-utils';
import { HomeIcon } from '../../../atoms';
import BottomTabBarBase from './BottomTabBar';

const Tab = createBottomTabNavigator();

const MockComponent = () => {
  return (
    <NavigationContainer>
      <Tab.Navigator
        screenOptions={({ route }) => ({
          tabBarTestID: `tab-${route.name}`,
        })}
        tabBar={props => <BottomTabBarBase {...props} />}
      >
        <Tab.Screen
          name="Home"
          component={() => null}
          options={{
            tabBarIcon: ({ size, color }) => (
              <HomeIcon width={size} height={size} fill={color} />
            ),
          }}
        />
        <Tab.Screen
          name="Accounts"
          component={() => null}
          options={{
            tabBarIcon: ({ size, color }) => (
              <HomeIcon width={size} height={size} fill={color} />
            ),
          }}
        />
        <Tab.Screen
          name="Profile"
          component={() => null}
          options={{
            tabBarIcon: ({ size, color }) => (
              <HomeIcon width={size} height={size} fill={color} />
            ),
          }}
        />
      </Tab.Navigator>
    </NavigationContainer>
  );
};

describe('BottomTabBarBase', () => {
  it('renders a tab bar with correct label and marker positions', async () => {
    const { comp, tabs } = renderTabs({
      elem: <MockComponent />,
      testIDs: ['tab-Home', 'tab-Accounts', 'tab-Profile', 'marker'],
    });
    await testTabs({ comp, tabs });
    expect(comp.toJSON()).toMatchSnapshot();
    expect(tabs[3].props.style.transform[0].translateX).toBe(0);
  });
});
