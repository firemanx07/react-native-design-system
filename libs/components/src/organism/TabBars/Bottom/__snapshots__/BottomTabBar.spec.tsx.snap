// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomTabBarBase renders a tab bar with correct label and marker positions 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    hasTwoStates={true}
    style={
      {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <View
      pointerEvents="auto"
      style={
        [
          {
            "flex": 1,
            "overflow": "hidden",
          },
          [
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            {
              "zIndex": 0,
            },
          ],
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="auto"
        removeClippedSubviews={false}
        style={
          {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            [
              {
                "backgroundColor": "rgb(242, 242, 242)",
                "flex": 1,
              },
              [
                {
                  "flex": 1,
                  "flexDirection": "column-reverse",
                },
                undefined,
              ],
            ]
          }
        >
          <View
            style={
              {
                "flex": 1,
              }
            }
          />
          <View
            onLayout={[Function]}
            style={null}
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "zIndex": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "rgb(255, 255, 255)",
                    "borderBottomColor": "rgb(216, 216, 216)",
                    "flex": 1,
                    "shadowColor": "rgb(216, 216, 216)",
                    "shadowOffset": {
                      "height": 0.5,
                      "width": 0,
                    },
                    "shadowOpacity": 0.85,
                    "shadowRadius": 0,
                  }
                }
              />
            </View>
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                {
                  "height": 44,
                  "maxHeight": undefined,
                  "minHeight": undefined,
                  "opacity": undefined,
                  "transform": undefined,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  {
                    "height": 0,
                  }
                }
              />
              <View
                pointerEvents="box-none"
                style={
                  {
                    "alignItems": "stretch",
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  collapsable={false}
                  pointerEvents="box-none"
                  style={
                    {
                      "alignItems": "flex-start",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "justifyContent": "center",
                      "marginStart": 0,
                    }
                  }
                />
                <View
                  collapsable={false}
                  pointerEvents="box-none"
                  style={
                    {
                      "justifyContent": "center",
                      "marginHorizontal": 16,
                      "maxWidth": 718,
                    }
                  }
                >
                  <Text
                    accessibilityRole="header"
                    aria-level="1"
                    collapsable={false}
                    numberOfLines={1}
                    style={
                      {
                        "color": "rgb(28, 28, 30)",
                        "fontSize": 17,
                        "fontWeight": "600",
                      }
                    }
                  >
                    Home
                  </Text>
                </View>
                <View
                  collapsable={false}
                  pointerEvents="box-none"
                  style={
                    {
                      "alignItems": "flex-end",
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "justifyContent": "center",
                      "marginEnd": 0,
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "backgroundColor": "#FFFFFF",
          "borderRadius": 7,
          "flexDirection": "row",
          "minHeight": 64,
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": true,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "minHeight": 35,
          "opacity": 1,
        }
      }
      testID="tab-Home"
    >
      <test-file-stub
        fill="#4D37D9"
        height={20}
        width={20}
      />
      <Text
        collapsable={false}
        style={
          {
            "color": "#2C2C2C",
            "fontSize": 14,
            "fontWeight": 700,
            "lineHeight": 20,
          }
        }
      >
        Home
      </Text>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "minHeight": 35,
          "opacity": 1,
        }
      }
      testID="tab-Accounts"
    >
      <test-file-stub
        fill="#C7C7D2"
        height={20}
        width={20}
      />
      <Text
        collapsable={false}
        style={
          {
            "color": "#2C2C2C",
            "fontSize": 14,
            "fontWeight": 400,
            "lineHeight": 20,
          }
        }
      >
        Accounts
      </Text>
    </View>
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "minHeight": 35,
          "opacity": 1,
        }
      }
      testID="tab-Profile"
    >
      <test-file-stub
        fill="#C7C7D2"
        height={20}
        width={20}
      />
      <Text
        collapsable={false}
        style={
          {
            "color": "#2C2C2C",
            "fontSize": 14,
            "fontWeight": 400,
            "lineHeight": 20,
          }
        }
      >
        Profile
      </Text>
    </View>
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#4D37D9",
          "borderRadius": 4,
          "height": 5,
          "position": "absolute",
          "top": 0,
          "transform": [
            {
              "translateX": 0,
            },
            {
              "translateX": -9.5,
            },
          ],
          "width": 19,
        }
      }
      testID="marker"
    />
  </View>
</RNCSafeAreaProvider>
`;
